gce/survival-training/dummy/train_test.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import glob
+import json
+import os
+import random
+import tempfile
+import unittest
+
+from . import train
+
+
+class TrainTest(unittest.TestCase):
+
+  def setUp(self):
+    self.job_dir = tempfile.mkdtemp()
+    self.num_checkpoints = 10
+    self.checkpoint_files = []
+    self.checkpoint_steps = 100
+
+    self.test_job_dir = tempfile.mkdtemp()
+    self.test_job_file_glob = os.path.join(self.test_job_dir, "*")
+
+    # Note that hyperparameters are intended to be constant across checkpoints
+    self.hyperparameter_1 = 17
+    self.hyperparameter_2 = 3.14159
+
+    for i in range(self.num_checkpoints):
```

Comment:
Convert this into list comprehensions

Reply:
There is no avoiding a for loop here (although it is possible to limit the amount of work that happens in the loop - `path` and `checkpoint_data` can be defined using comprehensions or `map` calls) - the `json.dump` call is not something one would want to map and represents a proper foreach.

Given that, I feel this makes the test clear.

Reply:
Yes I didn't notice the indent. Agree not worth it.